src/digitalIdentity-backend/src/test/java/didentity/amos/digitalIdentity/Services/AuthenticationServiceTest.java

```diff
@@ -0,0 +1,206 @@
+package didentity.amos.digitalIdentity.services;
+
+import org.junit.jupiter.api.*;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.util.Optional;
+
+import didentity.amos.digitalIdentity.enums.UserRole;
+import didentity.amos.digitalIdentity.model.User;
+import didentity.amos.digitalIdentity.repository.UserRepository;
+
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+@DirtiesContext(classMode = DirtiesContext.ClassMode.AFTER_EACH_TEST_METHOD)
+@ContextConfiguration(classes = {AuthenticationService.class})
+@ExtendWith(SpringExtension.class)
+public class AuthenticationServiceTest {
+    @Autowired
+    private AuthenticationService authenticationService;
+
+    @MockBean
+    private UserRepository userRepository;
+
+    ResponseEntity<String> response401;
+    ResponseEntity<String> response403;
+    ResponseEntity<String> lastError;
+
+    User user;
+
+    @BeforeEach
+    void setUp() {
+        response401 = ResponseEntity.status(401).body("Unauthorized, missing authentication.");
+        response403 = ResponseEntity.status(403).body("Forbidden.");
+        lastError = null;
+
+        user = new User();
+    }
+
+    @Test
+    void testAuthentication() {
+        assertFalse(authenticationService.authentication("John"));
```

Comment:
Maybe you can create two array at the top containing the authentication tokens (valid authentication + invalid authentication) and test the items. It might be better when we change the authentication tokens

Reply:
I tried to do it differently but it didn't work as I wanted so I have decided to keep it simple

Reply:
I understand it, but doesn't it work with just two string arrays? It's just because when we change the authentication tokens, you have to change all the occurrences in the test. If you have them in a string array and reference them you have to change them only once